how-to/customize-workspace/client/src/modules/composite/pages/menus.ts

```diff
@@ -0,0 +1,158 @@
+			const includeShowPage =
+				this._settings?.showPage?.include === undefined || this._settings?.showPage?.include;
+			const showPagesMenu: OpenFin.MenuItemTemplate[] = [];
+			const showPageMenuEntry: MenuEntry = {
+				label: this._settings?.showPage?.menuLabel ?? "Show Page",
+				icon: this._settings?.showPage?.menuIcon,
+				enabled: pages.length > 0,
+				submenu: [],
+				position: {
+					type: "SavePageAs",
+					operation: "after",
+					customId: "ShowPage",
+					...this._settings?.showPage?.menuPosition
+				}
+			};
+			const deletePageMenuEntry: MenuEntry = {
+				label: this._settings?.deletePage?.menuLabel ?? "Delete Page",
+				icon: this._settings?.deletePage?.menuIcon,
+				enabled: pages.length > 0,
+				submenu: [],
+				position: {
+					type: "SavePageAs",
+					operation: "after",
+					customId: "ShowDelete",
+					...this._settings?.deletePage?.menuPosition
+				}
+			};
+
+			const deletePagesMenu: OpenFin.MenuItemTemplate[] = [];
+
+			let browserWindowIdentity: OpenFin.Identity = relatedMenuId.windowIdentity;
+
+			const browserWindow = platform.Browser.wrapSync(browserWindowIdentity);
+
+			const options = await browserWindow.openfinWindow.getOptions();
+			const workspaceOptions: OpenFin.WorkspacePlatformOptions = options.workspacePlatform;
+
+			if (workspaceOptions.disableMultiplePages === true) {
+				browserWindowIdentity = undefined;
+			}
+
+			const allOpenPages = await platform.Browser.getAllAttachedPages();
+
+			if (pages.length > 0) {
+				for (const page of pages) {
+					const existingPage = allOpenPages.find((openPage) => page.pageId === openPage.pageId);
+					const isActiveExistingPageOnCurrentWindow =
+						existingPage !== undefined &&
+						existingPage.parentIdentity.name === browserWindowIdentity.name &&
+						existingPage?.isActive;
+					showPagesMenu.push({
+						label: page.title,
+						type: "normal",
+						enabled: !isActiveExistingPageOnCurrentWindow,
+						data: {
+							type: "Custom" as GlobalContextMenuOptionType.Custom,
+							action: {
+								id: existingPage !== undefined ? "page-show" : "page-open",
+								customData: {
+									pageId: page.pageId,
+									windowIdentity:
+										existingPage !== undefined ? existingPage.parentIdentity : browserWindowIdentity
+								}
+							}
+						}
+					});
+					deletePagesMenu.push({
+						label: page.title,
+						type: "normal",
+						enabled: existingPage === undefined,
+						data: {
+							type: "Custom" as GlobalContextMenuOptionType.Custom,
+							action: {
+								id: "page-delete",
+								customData: {
+									pageId: page.pageId
+								}
+							}
+						}
+					});
+				}
+				if (showPageMenuEntry.submenu) {
+					showPageMenuEntry.submenu.push(...showPagesMenu);
+				}
+				if (deletePageMenuEntry.submenu) {
+					deletePageMenuEntry.submenu.push(...deletePagesMenu);
+				}
+			}
+
+			const menuItemsToReturn = [];
```

Comment:
Can you just type this as MenuEntry[] to remove the eslint disable on 154

Reply:
Typing it still doesn't make it go away. I'll type it and explain why the rule is still disabled (possible bug in eslint).